src/Illuminate/Database/Query/Builder.php

```diff
@@ -1995,15 +1989,50 @@ protected function enforceOrderBy()
      */
     public function pluck($column, $key = null)
     {
-        $results = $this->get(is_null($key) ? [$column] : [$column, $key]);
+        $queryResult = $this->onceWithColumns(
+            is_null($key) ? [$column] : [$column, $key],
+            function () {
+                return $this->processor->processSelect($this, $this->runSelect());
+            }
+        );
+
+        if (empty($queryResult)) {
+            return collect();
+        }
 
         // If the columns are qualified with a table or have an alias, we cannot use
         // those directly in the "pluck" operations since the results from the DB
         // are only keyed by the column itself. We'll strip the table out here.
-        return $results->pluck(
-            $this->stripTableForPluck($column),
-            $this->stripTableForPluck($key)
-        );
+        $column = $this->stripTableForPluck($column);
+        $key = $this->stripTableForPluck($key);
+
+        if (is_array($queryResult[0])) {
+            return $this->pluckFromArrayColumn($queryResult, $column, $key);
+        } else {
```

Comment:
This `else` can be remove, as we already early returned something.

Reply:
I'm getting more and more annoyed by this trend. Also, it shouldn't be applied everywhere blindly. Sometimes it's not appropriate, for example here it's better to keep the if/else.